### PR TITLE
Feature/existing insurance + force to select selects

### PR DESCRIPTION
**Pull request by palmfjord**
like the title says  ☝️ This makes the flow go the whole way until the 'show offer', next steps are to actually get an offer based on the input and connect it to #8 🤘

### CODE REVIEW

File: src/pages/Chat/state.tsx [hunk 1]
```diff
@@ -120,6 +133,55 @@ export const ChatContainer: React.SFC<
                 },
               })
             },
+            setHasCurrentInsurance: (
+              event: React.ChangeEvent<HTMLSelectElement>,
+            ) => ({ state, setState }: EffectProps<State>) => {
+              const newState: Partial<State> = {
+                currentInsurance: cond([
```
Comment (hedvigoscar) [inline src/pages/Chat/state.tsx]: I'm having trouble reading this 😕

File: src/pages/Chat/steps/CurrentInsuranceInput.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import * as React from 'react'
+import * as yup from 'yup'
+import { ChatContainer, CurrentInsuranceState, Insurer } from '../state'
+
+interface CurrentInsuranceInputProps {
+  appear?: boolean
+  onSubmit?: () => void
+}
+
+const insurerNames = new Map<Insurer, string>([
```
Comment (hedvigoscar) [inline src/pages/Chat/steps/CurrentInsuranceInput.tsx]: We're missing a few here aren't we? Maybe not important for the scope of the PR though 😄

Comment (palmfjord) [reply src/pages/Chat/steps/CurrentInsuranceInput.tsx]: Indeed, idk which ones should be here but imo its kind of a ui thing

File: src/pages/Chat/state.tsx (hunk 1 above)
Comment (palmfjord) [reply src/pages/Chat/state.tsx]: ):

Comment (palmfjord) [reply src/pages/Chat/state.tsx]: ill refactor

Comment (palmfjord) [reply src/pages/Chat/state.tsx]: i have a PR to remove it completely coming up after these are merged. I see no reason to pull it in if we only use 4 functions from it.

File: src/pages/Chat/ChatConversation.tsx [hunk 3]
```diff
@@ -64,13 +65,27 @@ export const ChatConversation: React.SFC = () => (
 
         <Message id="current-insurance-question">
           {({ appear }) => (
-            <ChatMessage appear={appear}>
+            <ChatMessage
+              appear={appear}
+              onTyped={() => goToStep('current-insurance-input')}
+            >
               {livingSituation.numberOfPeople > 1
                 ? 'Trevligt! Hoppas ni trivs. Har du någon hemförsäkring redan?'
```
[the remark this answers is not in the document]
Comment (palmfjord) [reply src/pages/Chat/ChatConversation.tsx]: This needs to be merged be4 the text key pr can be merged.

File: src/pages/Chat/ChatConversation.tsx [hunk 4]
```diff
@@ -64,13 +65,27 @@ export const ChatConversation: React.SFC = () => (
...
                 : 'Trevligt! Hoppas du trivs. Har du någon hemförsäkring redan?'}
             </ChatMessage>
           )}
         </Message>
+
+        <Message id="current-insurance-input" delay={300}>
+          {({ appear }) => (
+            <CurrentInsuranceInput
+              appear={appear}
+              onSubmit={() => goToStep('show-offer')}
```
[the remark this answers is not in the document]
Comment (palmfjord) [reply src/pages/Chat/ChatConversation.tsx]: Sure, it's a good point! Maybe push it for later though?

[the remark this answers is not in the document]
Comment (hedvigoscar) [reply src/pages/Chat/ChatConversation.tsx]: I think we should scope that in, enums are awesome

File: src/pages/Chat/ChatConversation.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply src/pages/Chat/ChatConversation.tsx]: +1

File: src/pages/Chat/steps/CurrentInsuranceInput.tsx (hunk 2 above)
Comment (hedvigoscar) [reply src/pages/Chat/steps/CurrentInsuranceInput.tsx]: We can do the big four for now I think.

If, Folksam, Trygg-Hansa and Länsförsäkringar 🙂

**Comment by hedvigoscar** [comment]
✅ post enum and big four!

File: src/pages/Chat/ChatConversation.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Iamsamwhoami) [reply src/pages/Chat/ChatConversation.tsx]: oki let it be for now then :)

**Pull request merged**